File: src/main/java/br/com/g1bet/controller/TimeController.java
package br.com.g1bet.controller;

import br.com.g1bet.model.Time;
import br.com.g1bet.service.TimeService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/times")
@CrossOrigin("*")
public class TimeController {

    private final TimeService timeService;

    public TimeController(TimeService timeService) {
        this.timeService = timeService;
    }

    @GetMapping
    public ResponseEntity<List<Time>> buscarTodosTimes() {
        return ResponseEntity.ok(timeService.buscarTodosTimes());
    }

    @GetMapping("/{id}")
    public ResponseEntity<Time> buscarId(@PathVariable Long id) {
        return ResponseEntity.ok(timeService.buscarId(id));
    }

//    @GetMapping("/{time}")
//    public ResponseEntity<List<Time>> getByTime(@PathVariable String nome) {
//        return ResponseEntity.ok(repository.findAllByNomeContainingIgnoreCase(nome));
//    }

    @PostMapping
    public ResponseEntity<Time> post(@RequestBody Time time) {
        return ResponseEntity.status(HttpStatus.CREATED).body(timeService.cadastrar(time));
    }

    @PutMapping
    public ResponseEntity<Time> put(@RequestBody Time time, Long id) {
        return ResponseEntity.status(HttpStatus.OK).body(timeService.atualizar(time, id));
    }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable Long id) {
        timeService.deletar(id);
    }

}

File: src/main/java/br/com/g1bet/controller/TimeController.kt
package br.com.g1bet.controller

import br.com.g1bet.model.Time
import br.com.g1bet.service.TimeService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/times")
@CrossOrigin("*")
class TimeController(timeService: TimeService) {
    private val timeService: TimeService

    init {
        this.timeService = timeService
    }

    @GetMapping
    fun buscarTodosTimes(): ResponseEntity<List<Time>> {
        return ResponseEntity.ok(timeService.buscarTodosTimes())
    }

    @GetMapping("/{id}")
    fun buscarId(@PathVariable id: Long?): ResponseEntity<Time> {
        return ResponseEntity.ok(timeService.buscarId(id))
    }

    //    @GetMapping("/{time}")
    //    public ResponseEntity<List<Time>> getByTime(@PathVariable String nome) {
    //        return ResponseEntity.ok(repository.findAllByNomeContainingIgnoreCase(nome));
    //    }
    @PostMapping
    fun post(@RequestBody time: Time?): ResponseEntity<Time> {
        return ResponseEntity.status(HttpStatus.CREATED).body(timeService.cadastrar(time))
    }

    @PutMapping
    fun put(@RequestBody time: Time?, id: Long?): ResponseEntity<Time> {
        return ResponseEntity.status(HttpStatus.OK).body(timeService.atualizar(time, id))
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long?) {
        timeService.deletar(id)
    }
}